Content.Shared/Animals/UdderComponent.cs

```diff
@@ -0,0 +1,56 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.FixedPoint;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Animals;
+
+/// <summary>
+///     Gives the ability to produce a solution;
+///     produces endlessly if the owner does not have a HungerComponent.
+/// </summary>
+[RegisterComponent, AutoGenerateComponentPause, NetworkedComponent]
+public sealed partial class UdderComponent : Component
+{
+    /// <summary>
+    ///     The reagent to produce.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public ProtoId<ReagentPrototype> ReagentId = new();
+
+    /// <summary>
+    ///     The name of <see cref="Solution"/>.
+    /// </summary>
+    [DataField]
+    public string SolutionName = "udder";
+
+    /// <summary>
+    ///     The solution to add reagent to.
+    /// </summary>
+    public Entity<SolutionComponent>? Solution = null;
+
+    /// <summary>
+    ///     The amount of reagent to be generated on update.
+    /// </summary>
+    [DataField]
```

Comment:
```suggestion
    [DataField, AutoNetworkedField]
```
in case this is changed somehow